src/main/java/com/spotify/annoy/ANNIndex.java

```diff
@@ -8,53 +8,69 @@
 import java.util.*;
 
 /**
- * Read-only Approximate Nearest Neighbor Index which queries databases created by annoy.
+ * Read-only Approximate Nearest Neighbor Index which queries
+ * databases created by annoy.
  */
-public class ANNIndex implements AnnoyIndex {
+public class ANNIndex {
 
-  private int dimension, minLeafSize, nodeSize;
+  private static final int FLOAT_SIZE = 4;
+  private static final int INT_SIZE = 4;
```

Comment:
You can use the consts Integer.BYTES and Float.BYTES instead of making your own here